src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int currentCapacity = DEFAULT_CAPACITY;
+    private T[] elements = (T[]) new Object[DEFAULT_CAPACITY];
```

Comment:
Initialize the variable in the constructor
```suggestion
    private T[] elements;
```